distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2165,3 +2165,54 @@ async def test_set_index_p2p(c, s, *workers):
     await c.close()
     await asyncio.gather(*[check_worker_cleanup(w) for w in workers])
     await check_scheduler_cleanup(s)
+
+
+def test_shuffle_p2p_with_existing_index(loop):
+    df = pd.DataFrame({"a": np.random.randint(0, 3, 20)}, index=np.random.random(20))
+    ddf = dd.from_pandas(
+        df,
+        npartitions=4,
+    )
+    with LocalCluster(
+        n_workers=2, dashboard_address=":0", loop=loop
+    ) as cluster, Client(cluster) as c:
+        ddf = ddf.shuffle("a", shuffle="p2p")
+        result = c.compute(ddf, sync=True)
+        dd.assert_eq(result, df)
+
+
+def test_set_index_p2p_with_existing_index(loop):
+    df = pd.DataFrame({"a": np.random.randint(0, 3, 20)}, index=np.random.random(20))
+    ddf = dd.from_pandas(
+        df,
+        npartitions=4,
+    )
+    with LocalCluster(
+        n_workers=2, dashboard_address=":0", loop=loop
+    ) as cluster, Client(cluster) as c:
+        ddf.set_index("a", shuffle="p2p")
```

Comment:
Why are you not using the result of this op?

Reply:
Whoopsie, changed this test too many times.

Reply:
Fixed